infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java

```diff
@@ -63,22 +126,67 @@ public KubernetesClientFactory(
     if (doTrustCerts != null) {
       configBuilder.withTrustCerts(doTrustCerts);
     }
-    this.client = new UnclosableKubernetesClient(configBuilder.build());
+
+    Config config = configBuilder.build();
+    return config;
   }
 
   /**
    * Creates instance of {@link KubernetesClient}.
    *
    * @throws InfrastructureException if any error occurs on client instance creation.
    */
-  public KubernetesClient create() throws InfrastructureException {
-    return client;
+  private KubernetesClient create(Config config) throws InfrastructureException {
+    OkHttpClient clientHttpClient =
+        httpClient.newBuilder().authenticator(Authenticator.NONE).build();
+    OkHttpClient.Builder builder = clientHttpClient.newBuilder();
+    builder.interceptors().clear();
+    clientHttpClient =
+        builder
+            .addInterceptor(
+                new Interceptor() {
+                  @Override
+                  public Response intercept(Chain chain) throws IOException {
+                    Request request = chain.request();
+                    if (isNotNullOrEmpty(config.getUsername())
```

Comment:
If I'm not mistaken this code is copied from HttpClientUtils. If so, it is an incomplete copy. Can you elaborate why don't we use HttpClientUtils instead of this code? If there is no specific reason I would suggest using original code. It is less error-prone in case of any further changes in k8s API.

Reply:
We only want to apply the part that is related to the Authentication interceptor. All the other common configuration steps have already been applied when creating the shared base `OkHttpClient` by calling precisely the `HttpClientUtils.createHttpClient()` method. But here we only want to override the authentication interceptor.

Reply:
I think it is very unclear that this takes place. Can you add comments to the code that explains this behavior?

Reply:
I added comments in commit https://github.com/eclipse/che/pull/8805/commits/2f9d7cf6e0fc28b099f36180b4c4303569e9ca43